### ISSUE TITLE
Neovide crashes immediately when openning. | Windows 10

### ISSUE THREAD

**Issue by 0xbalazstoth** [neovide/neovide#907]
**Describe the bug**
Basically when I open neovide.exe, immediately crashes.

**Expected behavior**
To work and not crash.

**Screenshots**
![image](https://user-images.githubusercontent.com/12668764/130283486-cb04c40e-1104-403b-9244-f0b84fbfbcc5.png)

@rust_panic error
```asm
; id = {0x00008394}, range = [0x00000001406bab80-0x00000001406bac71), name="rust_panic"
; Source location: \rustc\a178d0322ce20e33eac124758e837cbd80a6f633\library\std\src\panicking.rs:667
7FF7D70FAB80: 48 81 EC C8 00 00 00       subq   $0xc8, %rsp
7FF7D70FAB87: 48 89 4C 24 28             movq   %rcx, 0x28(%rsp)
7FF7D70FAB8C: 48 89 54 24 30             movq   %rdx, 0x30(%rsp)
7FF7D70FAB91: 48 8D 4C 24 28             leaq   0x28(%rsp), %rcx
7FF7D70FAB96: E8 35 AD 00 00             callq  0x7ff7d71058d0  ; void panic_unwind::__rust_start_panic() at lib.rs:103
7FF7D70FAB9B: 89 44 24 24                movl   %eax, 0x24(%rsp)
7FF7D70FAB9F: 48 8D 44 24 24             leaq   0x24(%rsp), %rax
7FF7D70FABA4: 48 89 44 24 58             movq   %rax, 0x58(%rsp)
7FF7D70FABA9: 48 8D 05 50 F2 01 00       leaq   0x1f250(%rip), %rax  ; void core::fmt::num::imp::{{impl}}::fmt() at num.rs:279
7FF7D70FABB0: 48 89 44 24 60             movq   %rax, 0x60(%rsp)
7FF7D70FABB5: 48 8D 05 5C 26 CA 00       leaq   0xca265c(%rip), %rax
7FF7D70FABBC: 48 89 84 24 98 00 00 00    movq   %rax, 0x98(%rsp)
7FF7D70FABC4: 48 C7 84 24 A0 00 00 00 >  movq   $0x1, 0xa0(%rsp)
7FF7D70FABD0: 48 C7 84 24 A8 00 00 00 >  movq   $0x0, 0xa8(%rsp)
7FF7D70FABDC: 48 8D 44 24 58             leaq   0x58(%rsp), %rax
7FF7D70FABE1: 48 89 84 24 B8 00 00 00    movq   %rax, 0xb8(%rsp)
7FF7D70FABE9: 48 C7 84 24 C0 00 00 00 >  movq   $0x1, 0xc0(%rsp)
7FF7D70FABF5: 48 8D 84 24 98 00 00 00    leaq   0x98(%rsp), %rax
7FF7D70FABFD: 48 89 44 24 48             movq   %rax, 0x48(%rsp)
7FF7D70FAC02: 48 8D 05 57 94 01 00       leaq   0x19457(%rip), %rax  ; void core::fmt::{{impl}}::fmt() at mod.rs:466
7FF7D70FAC09: 48 89 44 24 50             movq   %rax, 0x50(%rsp)
7FF7D70FAC0E: 48 8D 05 03 20 CA 00       leaq   0xca2003(%rip), %rax
7FF7D70FAC15: 48 89 44 24 68             movq   %rax, 0x68(%rsp)
7FF7D70FAC1A: 48 C7 44 24 70 02 00 00 >  movq   $0x2, 0x70(%rsp)
7FF7D70FAC23: 48 C7 44 24 78 00 00 00 >  movq   $0x0, 0x78(%rsp)
7FF7D70FAC2C: 48 8D 44 24 48             leaq   0x48(%rsp), %rax
7FF7D70FAC31: 48 89 84 24 88 00 00 00    movq   %rax, 0x88(%rsp)
7FF7D70FAC39: 48 C7 84 24 90 00 00 00 >  movq   $0x1, 0x90(%rsp)
7FF7D70FAC45: 48 8D 4C 24 20             leaq   0x20(%rsp), %rcx
7FF7D70FAC4A: 48 8D 54 24 68             leaq   0x68(%rsp), %rdx
7FF7D70FAC4F: E8 6C 60 FF FF             callq  0x7ff7d70f0cc0  ; static void std::io::Write::write_fmt<std::sys::windows::stdio::Stderr>() at mod.rs:1567
7FF7D70FAC54: 48 89 54 24 40             movq   %rdx, 0x40(%rsp)
7FF7D70FAC59: 48 89 44 24 38             movq   %rax, 0x38(%rsp)
7FF7D70FAC5E: 48 8D 4C 24 38             leaq   0x38(%rsp), %rcx
7FF7D70FAC63: E8 88 9A FE FF             callq  0x7ff7d70e46f0  ; static void core::ptr::drop_in_place<enum$<core::result::Result<tuple<>, std::io::error::Error>, 0, 3, Err>>() at mod.rs:192
7FF7D70FAC68: B9 07 00 00 00             movl   $0x7, %ecx
7FF7D70FAC6D: CD 29                      int    $0x29
7FF7D70FAC6F: 0F 0B                      ud2    
```

**Desktop (please complete the following information):**
 - OS: Windows 10 21H1
 - Neovide Version 0.8.0.0
 - Neovim Version 0.6
 - GPU: AMD Radeon RX 580 8GB

**Please run `neovide --log` and paste the contents of the `.log` file here:**
```
TRACE [neovide] Neovide version: 0.8.0
INFO [neovide::bridge] Starting neovim with: Command { std: "C:\\tools\\neovim\\Neovim\\bin\\nvim.EXE" "--embed", kill_on_drop: false }
INFO [neovide::bridge] Close watcher started
DEBUG [neovide::bridge::events] Ignored client type property: "methods"
DEBUG [neovide::bridge::events] Ignored client type property: "attributes"
INFO [neovide::bridge] Neovide registered to nvim with channel id 1
INFO [neovide::bridge] Neovim process attached
```

**Comment by Kethku** [comment]
At the very least we should be able to print a more useful error here. Its crazy that nothing showed up on windows

**Comment by DiegoMagdaleno** [comment]
Seems like it could be an issue with AMD graphic cards?:
https://github.com/rust-windowing/glutin/issues/1219

[Rust windowing](https://github.com/rust-windowing/glutin/blob/92a3ad375555cbcf6a0fc20d8aaa782bcb3b6656/ISSUES.md)

**Comment by shift-d** [comment]
I have same error message in #825

**Comment by dartacao** [comment]
I have a feeling this is the exact same issue I have with neovide... crashing on startup with an AMD RX480, no useful logs from --log.

**Comment by DiegoMagdaleno** [comment]
This looks like an AMD issue.

**Comment by dartacao** [comment]
> This looks like an AMD issue.

Just checked on my laptop with latest. Besides weird ass font rendering (which I should make an issue out of, when I have the time), no crashes. Laptop has no GPU, just an Intel i5 on integrated.

**Comment by leon332157** [comment]
I have a RX580 with the latest 21.8.1 adrenaline driver, no issues here.

**Comment by 0xbalazstoth** [comment]
> I have a RX580 with the latest 21.8.1 adrenaline driver, no issues here.

I've 21.7.2, just got the latest version, let me update, fingers crossed

**Comment by leon332157** [comment]
Just as a side note, I saw the error message in the screenshot, I suggest everyone with the issue check the "pixel format" option in the Radeon Software, as shown. That could be the problem.
![image](https://user-images.githubusercontent.com/30728609/130345964-0ad1e27f-65db-4476-bcec-599a451fded9.png)

**Comment by 0xbalazstoth** [comment]
Updated and this "pixel format" option is default. :(

**Comment by leon332157** [comment]
Can you check the "Advanced display settings" in windows display settings, might not be too much of a help but just wanted to check. I think maybe changing some settings could help.
![image](https://user-images.githubusercontent.com/30728609/130368348-7e6e5c8b-13a7-4521-8d95-bff0629a2d42.png)

This is what mine looks like:
![image](https://user-images.githubusercontent.com/30728609/130368362-ab6ce876-1bea-4518-92cd-cab24d2443b0.png)

**Comment by 0xbalazstoth** [comment]
> Can you check the "Advanced display settings" in windows display settings, might not be too much of a help but just wanted to check. I think maybe changing some settings could help.
> ![image](https://user-images.githubusercontent.com/30728609/130368348-7e6e5c8b-13a7-4521-8d95-bff0629a2d42.png)
> 
> This is what mine looks like:
> ![image](https://user-images.githubusercontent.com/30728609/130368362-ab6ce876-1bea-4518-92cd-cab24d2443b0.png)

same as yours, but my resolution is 1920x1080

**Comment by leon332157** [comment]
Ok, I’m guessing the only solution to this problem is changing the format in the code then, maybe you can try change the format and building again?

**Comment by 0xbalazstoth** [comment]
> Ok, I’m guessing the only solution to this problem is changing the format in the code then, maybe you can try change the format and building again?
> […](#)
> On Aug 22, 2021, at 16:02, Balázs Tóth ***@***.***> wrote: ﻿ Can you check the "Advanced display settings" in windows display settings, might not be too much of a help but just wanted to check. I think maybe changing some settings could help. This is what mine looks like: same as yours, but my resolution is 1920x1080 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

What do you mean by changing format? Color format?

**Comment by leon332157** [comment]
[here](https://github.com/neovide/neovide/blob/main/src/window/window_wrapper/mod.rs#L258) is the line of code
[documentation here](https://docs.rs/glutin/0.27.0/glutin/struct.ContextBuilder.html#method.with_pixel_format)

It seems like the pixel format being used is written in the code. I think you can try experimenting with this a little to see if that helps. 

Can you also check "display adapter properties" and "list all modes"
![image](https://user-images.githubusercontent.com/30728609/130368743-49a11d38-1598-4c94-a914-43ca466342ac.png)
I am on a 32 bit mode. 
![image](https://user-images.githubusercontent.com/30728609/130368749-9dce689f-0a98-4e80-8016-0fdbf1e0863a.png)

Could it be that, since we are asking for a 24 bit mode, but the GPU or display doesn't support that?

**Comment by dartacao** [comment]
> ```rust
> let windowed_context = ContextBuilder::new()
>         .with_pixel_format(24, 8)
>         .with_stencil_buffer(8)
>         .with_gl_profile(GlProfile::Core)
>         .with_vsync(false)
>         .with_srgb(true)
>         //.with_hardware_acceleration(None)
>         .build_windowed(winit_window_builder, &event_loop)
>         .unwrap();
> ```

Can confirm, this fixes it for me as well.

**Comment by farra** [comment]
Can confirm I also saw this issue and that the` with_srgb` to `true` worked on a fresh install of Windows 10 Pro on a brand new Asus ROG G15 (Ryzen 9).

**Comment by jesselawson** [comment]
For what it's worth, on a Windows 10 machine I did a fresh `git clone` and `cargo build --release` this morning and Neovide is running just fine.

**Comment by Kethku** [comment]
https://github.com/neovide/neovide/pull/944

Heres a PR which enables configuring the srgb window setting. Can somebody here try this?

**Comment by jesselawson** [comment]
> 
> 
> #944
> 
> Heres a PR which enables configuring the srgb window setting. Can somebody here try this?

Just downloaded this branch on Windows 10 Home, ran `cargo build --release`, and it built without error.

**Comment by Kethku** [comment]
I just merged #944 so I believe this is fixed. Please reopen if issues persist.
